gtsfm/utils/transforms.py

```diff
@@ -0,0 +1,48 @@
+"""Utility functions for geometric transformations.
+
+Authors: Akshay Krishnan
+"""
+from typing import List
+
+import numpy as np
+from gtsam import Unit3
+
+
+def euclidean_to_spherical_directions(directions: List[Unit3]) -> np.ndarray:
+    """Converts a list of Unit3 directions to spherical coordinates (azimuth, zenith).
+    Angles are given in a compass frame:
+    zenith=0 is along +y, azimuth=0 is along -z, and azimuth=pi/2 is along +x.
+    This follows 1DSfM's convention
+    https://github.com/wilsonkl/SfM_Init/blob/f801a4ace3b34f990cbda3c57b96387ce19c90c1/sfminit/onedsfm.py#L132
```

Comment:
can we use the latest develop/master branch in the URL, instead of this checksum, to make this more concise?

Reply:
There is a reason why we use permalink, the master can change but this link will always point to the code we want it to point to. Unless the repository is deleted of course.